model/QueueBroker/NewSqlQueueBroker.php

```diff
@@ -0,0 +1,271 @@
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoTaskQueue\model\QueueBroker;
+
+use common_persistence_Manager;
+use common_persistence_SqlPersistence;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\SchemaException;
+use Exception;
+use InvalidArgumentException;
+use oat\generis\Helper\UuidPrimaryKeyTrait;
+use oat\generis\persistence\PersistenceManager;
+use oat\tao\model\taskQueue\Queue\Broker\AbstractQueueBroker;
+use oat\tao\model\taskQueue\Task\TaskInterface;
+use oat\taoTaskQueue\model\QueueBroker\storage\NewSqlSchema;
+use Throwable;
+
+/**
+ * Storing messages/tasks in DB.
+ *
+ * @author Gyula Szucs <gyula@taotesting.com>
+ */
+class NewSqlQueueBroker extends AbstractQueueBroker
+{
+    use UuidPrimaryKeyTrait;
+
+    private $persistenceId;
+
+    /** @var common_persistence_SqlPersistence */
+    protected $persistence;
+
+    public function __construct(string $persistenceId, int $receiveTasks = 1)
+    {
+        parent::__construct($receiveTasks);
+
+        if (empty($persistenceId)) {
+            throw new InvalidArgumentException("Persistence id needs to be set for " . __CLASS__);
+        }
+
+        $this->persistenceId = $persistenceId;
+    }
+
+    public function __toPhpCode()
+    {
+        return 'new ' . get_called_class() . '('
+            . \common_Utils::toHumanReadablePhpString($this->persistenceId)
+            . ', '
+            . \common_Utils::toHumanReadablePhpString($this->getNumberOfTasksToReceive())
+            . ')';
+    }
+
+    protected function getPersistence(): ?common_persistence_SqlPersistence
+    {
+        if (is_null($this->persistence)) {
+            $this->persistence = $this->getServiceLocator()
+                ->get(common_persistence_Manager::SERVICE_ID)
+                ->getPersistenceById($this->persistenceId);
+        }
+
+        return $this->persistence;
+    }
+
+    private function getTableName(): string
+    {
+        return strtolower($this->getQueueNameWithPrefix());
+    }
+
+    /**
+     * Note: this method can be run multiple times because only the migrate queries (result of getMigrateSchemaSql) will be run.
+     *
+     * @inheritdoc
+     */
+    public function createQueue()
```

Comment:
As we agreed to always move public methods before private / protected. Can you please adjust the methods order? :)

Reply:
done